File: fstp-python/com/purefun/fstp/core/bo/otw/TestBO2_OTW.py
from core.common.ICommon_OTW import ICommon_OTW
from com.purefun.fstp.core.bo.pro import TestBO2_pb2
from com.purefun.fstp.core.bo.model.TestBO2 import TestBO2

class TestBO2_OTW(ICommon_OTW):

    def __init__(self, byteMsg = None):
        self._bo_pro = TestBO2_pb2.TestBO2()
        self._bo = TestBO2()

        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()

    def __setDataFromBO(self):
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.destination = self._bo.destination
        self._bo_pro.servername = self._bo.servername
        self._bo_pro.msg = self._bo.msg

    def __setDataFromPB(self):
        self._bo.uuid = self._bo_pro.uuid
        self._bo.boid = self._bo_pro.boid
        self._bo.destination = self._bo_pro.destination
        self._bo.servername = self._bo_pro.servername
        self._bo.msg = self._bo_pro.msg

    def getBO(self):
        return self._bo

    def getProBO(self):
        return self._bo_pro

    def getUuid(self):
        return self._bo.uuid

    def setUuid(self, uuid):
        self._bo.uuid = uuid
        self._bo_pro.uuid = uuid

    def getBoid(self):
        return self._bo.boid

    def setBoid(self, boid):
        self._bo.boid = boid
        self._bo_pro.boid = boid

    def getDestination(self):
        return self._bo.destination

    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination

    def getServername(self):
        return self._bo.servername

    def setServername(self, servername):
        self._bo.servername = servername
        self._bo_pro.servername = servername

    def getMsg(self):
        return self._bo.msg

    def setMsg(self, msg):
        self._bo.msg = msg
        self._bo_pro.msg = msg

    def toString(self):
        return "TestBO2_OTW ["+"uuid = " + str(self.getUuid()) +"," +"boid = " + str(self.getBoid()) +"," +"destination = " + str(self.getDestination()) +"," +"servername = " + str(self.getServername()) +"," +"msg = " + str(self.getMsg()) +"," +"]"

File: fstp-core-bo/resource/com/purefun/fstp/core/bo/otw/TestBO2_OTW.java
package com.purefun.fstp.core.bo.otw;

import com.purefun.fstp.core.bo.TestBO2;
import com.purefun.fstp.core.bo.pro.TestBO2_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fstp.core.bo.commom.ICommom_OTW;
import com.google.protobuf.Any;

public class TestBO2_OTW implements ICommom_OTW {
    TestBO2_PRO.TestBO2.Builder builder = null;
    TestBO2 bo = null;

    public TestBO2_OTW() {
        builder = TestBO2_PRO.TestBO2.newBuilder();
        bo= new TestBO2();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public TestBO2_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = TestBO2_PRO.TestBO2.newBuilder();
        bo= new TestBO2();
        TestBO2_PRO.TestBO2 receive = TestBO2_PRO.TestBO2.parseFrom(message);
        setServername(receive.getServername());
        setMsg(receive.getMsg());
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
    }

    public TestBO2_OTW(TestBO2 bofrom){
        builder = TestBO2_PRO.TestBO2.newBuilder();
        bo= new TestBO2();
        setServername(bofrom.servername);
        setMsg(bofrom.msg);
        setUuid(bofrom.uuid);
        setBoid(bofrom.boid);
        setDestination(bofrom.destination);
    }

    public byte[] serial() {
        return builder.build().toByteArray();
    }

    @Override
    public com.google.protobuf.GeneratedMessageV3.Builder getBuilder() { 
        return builder;
    }

    @Override
    public TestBO2 getBo() { 
        return bo;
    }

    public java.lang.String getServername() {
        return builder.getServername();
    }

    public void setServername(java.lang.String servername) {
        bo.servername = servername;
        builder.setServername(servername);
    }

    public java.lang.String getMsg() {
        return builder.getMsg();
    }

    public void setMsg(java.lang.String msg) {
        bo.msg = msg;
        builder.setMsg(msg);
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public String toString() {
        return "TestBO2_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "servername = " + getServername() +"," +
            "msg = " + getMsg() +"," +
         "]";
    }
}
